src/components/PopoverWithMeasuredContent.js

```diff
@@ -59,138 +60,120 @@ const defaultProps = {
+function PopoverWithMeasuredContent(props) {
+    const {windowWidth, windowHeight} = useWindowDimensions();
+    const [popoverWidth, setPopoverWidth] = useState(props.popoverDimensions.width);
+    const [popoverHeight, setPopoverHeight] = useState(props.popoverDimensions.height);
+    const [isContentMeasured, setIsContentMeasured] = useState(popoverWidth > 0 && popoverHeight > 0);
+    const [isVisible, setIsVisible] = useState(false);
 
     /**
      * When Popover becomes visible, we need to recalculate the Dimensions.
      * Skip render on Popover until recalculations have done by setting isContentMeasured false as early as possible.
-     *
-     * @static
-     * @param {Object} props
-     * @param {Object} state
-     * @return {Object|null}
      */
-    static getDerivedStateFromProps(props, state) {
-        // When Popover is shown recalculate
-        if (!state.isVisible && props.isVisible) {
-            return {isContentMeasured: lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0, isVisible: true};
-        }
-        if (!props.isVisible) {
-            return {isVisible: false};
-        }
-        return null;
+    if (!isVisible && props.isVisible) {
+        // We use additional function to guarantee that async state change would be completed
+        setIsVisible(() => { 
+            setIsContentMeasured(props.popoverDimensions.width > 0 && props.popoverDimensions.height > 0);
+            return true;
+        });
     }
 
-    shouldComponentUpdate(nextProps, nextState) {
-        if (this.props.isVisible && (nextProps.windowWidth !== this.props.windowWidth || nextProps.windowHeight !== this.props.windowHeight)) {
-            return true;
+    useEffect(() => {
+        if (!props.isVisible) {
+            setIsVisible(false);
         }
-
-        // This component does not require re-render until any prop or state changes as we get the necessary info
-        // at first render. This component is attached to each message on the Chat list thus we prevent its re-renders
-        return !_.isEqual(_.omit(this.props, ['windowWidth', 'windowHeight']), _.omit(nextProps, ['windowWidth', 'windowHeight'])) || !_.isEqual(this.state, nextState);
-    }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.popoverDimensions.width, props.popoverDimensions.height, props.isVisible, isVisible]);
 
     /**
      * Measure the size of the popover's content.
      *
      * @param {Object} nativeEvent
      */
-    measurePopover({nativeEvent}) {
-        this.popoverWidth = nativeEvent.layout.width;
-        this.popoverHeight = nativeEvent.layout.height;
-        this.setState({isContentMeasured: true});
-    }
+    const measurePopover = ({nativeEvent}) => {
+        setIsContentMeasured(() => {
+            setPopoverWidth(nativeEvent.layout.width);
+            setPopoverHeight(nativeEvent.layout.height);
+            return true;
+        });
```

Comment:
This seems useless. Just call the functions one by one.

Reply:
@s77rt This form is following @luacmartins suggestion that state would be changed asynchronously https://github.com/Expensify/App/pull/22868#discussion_r1263874311

Reply:
That change does not change the fact that it's still async. The function form you are using is not a callback either.

Reply:
@s77rt Yes, but if we use a func we could be sure it will be called last in queue and right after setPopover...
https://react.dev/learn/queueing-a-series-of-state-updates

Reply:
I don't see any diff. Even if we call the functions one by one the order in the queue will be the same. Actually the order shouldn't matter anyway. React will batch those state updates and update them at once.

> React waits until all code in the event handlers has run before processing your state updates.